### PR TITLE
YD-546 Added device-specific links

### PR DESCRIPTION
**Pull request by Bert-R**
Several device-specific links that were already present on the User resource are now added to the Device resource. Once the apps are updated to take the link from the device, the links should be removed from the user (see YD-544).

This includes the following links:
* ``appleMobileConfig``
* ``postOpenAppEvent``
* ``appActivity``
* ``sslRootCert``

The same applies to one property:
* ``sslRootCertCN``

Along with it:
* Made the usage message of ``HandleDansGuardianLog.pl`` complete by adding ``--verbose``
* Made the name of the second device in ``ActivityTest.'Richard works on two devices'()`` less confusing

### CODE REVIEW

File: appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java [hunk 1]
```diff
@@ -397,18 +460,30 @@ public DeviceResource toResource(DeviceBaseDto device)
 			ControllerLinkBuilder selfLinkBuilder = getSelfLinkBuilder(device.getId());
 			addSelfLink(selfLinkBuilder, deviceResource);
 			addEditLink(selfLinkBuilder, deviceResource);
+			if (isRequestingDevice(device))
+			{
+				addPostOpenAppEventLink(deviceResource);
+				addAppActivityLink(deviceResource);
+				addSslRootCertificateLink(deviceResource);
```
Comment (aristotelos) [inline appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java]: ❓  This link is actually not different per device, is it? So why move it here?

File: appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java [hunk 2]
```diff
@@ -350,9 +385,35 @@ public static ControllerLinkBuilder getRegisterDeviceLinkBuilder(UUID userId)
...
+		}
+
+		public boolean isRequestingDevice()
+		{
+			return isRequestingDevice;
+		}
+
+		@JsonProperty("sslRootCertCN")
+		@JsonInclude(Include.NON_EMPTY)
+		public Optional<String> getSslRootCertCn()
```
Comment (aristotelos) [inline appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java]: ❓  This is not different per device too, is it?

File: appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java [hunk 3]
```diff
@@ -397,18 +460,30 @@ public DeviceResource toResource(DeviceBaseDto device)
 			ControllerLinkBuilder selfLinkBuilder = getSelfLinkBuilder(device.getId());
 			addSelfLink(selfLinkBuilder, deviceResource);
 			addEditLink(selfLinkBuilder, deviceResource);
+			if (isRequestingDevice(device))
```
Comment (aristotelos) [inline appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java]: ❓ [Initially](https://wiki.yona.nu/display/DEV/Adding+device+notion?focusedCommentId=18284704#comment-18284704), you were doubting if we should move the links at all. What is the reason that you think it is better to move them after all? It introduces some complexity to requesting the devices collection and marking the requesting device.

File: appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java (hunk 1 above)
Comment (Bert-R) [reply appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java]: Though it's not specific for the device, it's really device information. I want to group all such information on the device resource.

File: appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java (hunk 2 above)
Comment (Bert-R) [reply appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java]: See previous comment.

File: appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java (hunk 3 above)
Comment (Bert-R) [reply appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java]: It's a cleaner design. The information is really device specific, so I feel it's better to put the information in the context it belongs. It's more clear to the app builders.

File: appservice/src/intTest/groovy/nu/yona/server/AddDeviceTest.groovy [hunk 4]
```diff
@@ -86,18 +86,18 @@ class AddDeviceTest extends AbstractAppServiceIntegrationTest
...
-		assertDefaultOwnDevice(defaultDevice)
+		assertDefaultOwnDevice(defaultDevice, false)
 		assert newDevice.name == newDeviceName
 		assert newDevice.operatingSystem == newDeviceOs
 		assertDateTimeFormat(newDevice.registrationTime)
 		assertDateFormat(newDevice.appLastOpenedDate)
 
 		// User self-link uses the new device as "requestingDeviceId"
-		def idOfNewDevice = newDevice._links.self.href - ~/.*\//
+		def idOfNewDevice = newDevice._links.self.href - ~/.*\// - ~/\?.*/
```
[the remark this answers is not in the document]
Comment (Bert-R) [reply appservice/src/intTest/groovy/nu/yona/server/AddDeviceTest.groovy]: OK, will do.

**Comment by Bert-R** [comment]
Answered the questions and pushed an update.

**Pull request merged**